src/elements/content-uploader/ContentUploader.js

```diff
@@ -808,8 +817,11 @@ class ContentUploader extends Component<Props, State> {
             onUpload(item.boxFile);
         }
 
-        this.updateViewAndCollection(items);
-        this.upload();
+        this.updateViewAndCollection(items, () => {
+            if (view === VIEW_UPLOAD_IN_PROGRESS) {
+                this.upload();
+            }
```

Comment:
Same here, would this view be the most current view since it is being called AFTER the render?

Reply:
Good question... I'm going to make it clearer